src/burnchains/bitcoin/indexer.rs

```diff
@@ -104,155 +117,240 @@ pub struct BitcoinIndexer {
-            first_block: FIRST_BLOCK_MAINNET
-        };
+            spv_headers_path: spv_headers_path,
+            first_block: FIRST_BLOCK_MAINNET,
+            magic_bytes: BLOCKSTACK_MAGIC_MAINNET.clone()
+        }
     }
 
-    fn from_file(path: &str) -> Result<BitcoinIndexerConfig, &'static str> {
+    pub fn to_file(&self, path: &String) -> Result<(), btc_error> {
+       let mut conf = Ini::new();
+
+       let username = self.username.clone().unwrap_or("".to_string());
+       let password = self.password.clone().unwrap_or("".to_string());
+
+       conf.with_section(Some("bitcoin".to_owned()))
+           .set("server", self.peer_host.as_str())
+           .set("port", format!("{}", self.peer_port).as_str())
+           .set("rpc_port", format!("{}", self.rpc_port).as_str())
+           .set("rpc_ssl", format!("{}", self.rpc_ssl).as_str())
+           .set("username", username.as_str())
+           .set("password", password.as_str())
+           .set("timeout", format!("{}", self.timeout).as_str())
+           .set("spv_path", self.spv_headers_path.as_str())
+           .set("first_block", format!("{}", self.first_block).as_str());
+
+       conf.with_section(Some("blockstack".to_owned()))
+           .set("network_id", format!("{}{}", self.magic_bytes.as_bytes()[0] as char, self.magic_bytes.as_bytes()[1] as char).as_str());
+
+       conf.write_to_file(&path)
+           .map_err(|e| btc_error::Io(e))
+    }
+
+    pub fn from_file(path: &String) -> Result<BitcoinIndexerConfig, btc_error> {
        let conf_path = PathBuf::from(path);
        if !conf_path.is_file() {
-           return Err("Failed to load BitcoinIndexerConfig file: No such file or directory");
+           return Err(btc_error::ConfigError("Failed to load BitcoinIndexerConfig file: No such file or directory".to_string()));
        }
-       let default_config = BitcoinIndexerConfig::default();
+
+       let mut home_pathbuf = PathBuf::from(dirs::home_dir().unwrap());
+       home_pathbuf.push(".stacks");
+
+       let default_working_dir = home_pathbuf.to_str().unwrap().to_string();
+       let default_config = BitcoinIndexerConfig::default(&default_working_dir);
 
        match Ini::load_from_file(path) {
            Ok(ini_file) => {
                // got data!
                let bitcoin_section_opt = ini_file.section(Some("bitcoin").to_owned());
                if None == bitcoin_section_opt {
-                   return Err("No [bitcoin] section in config file");
+                   return Err(btc_error::ConfigError("No [bitcoin] section in config file".to_string()));
                }
 
                let bitcoin_section = bitcoin_section_opt.unwrap();
 
-               // defaults
+               // [bitcoin]
                let peer_host = bitcoin_section.get("server")
                                               .unwrap_or(&default_config.peer_host);
 
                let peer_port = bitcoin_section.get("p2p_port")
```

Comment:
This function is using `"p2p_port"` and `"port"` for the p2p port and rpc ports, the `to_file` function uses `"port"` and `"rpc_port"`

Reply:
Yup, caught that already.